Bottom: 7f027129a6843723f7b385c217f35c825c082c3c
Top:    43549522befa451661f22b1496073e501345f44b
Author: Ian Kent <raven@themaw.net>
Date:   2013-07-30 13:41:34 +0800

Refresh of autofs-5.0.7-add-dumpmap-to-file.patch

---

diff --git a/lib/master.c b/lib/master.c
index 37046a3..22a289e 100644
--- a/lib/master.c
+++ b/lib/master.c
@@ -1423,11 +1423,15 @@ int master_show_mounts(struct master *master, const char *maps)
 			lookup_prune_cache(ap, now);
 		else {
 			printf("  failed to read map\n\n");
+			if (map_name)
+				free(map_name);
 			continue;
 		}
 
 		if (!this->maps) {
 			printf("  no map sources found\n\n");
+			if (map_name)
+				free(map_name);
 			continue;
 		}
 
@@ -1476,6 +1480,9 @@ int master_show_mounts(struct master *master, const char *maps)
 				} while ((me = cache_lookup_next(source->mc, me)));
 			}
 next:
+			if (map_name)
+				free(map_name);
+
 			count++;
 
 			source = source->next;
